### ISSUE TITLE
T3 - SVM y tiempo de entrenamiento y mucha confusion

### ISSUE THREAD

**Issue by EleaTriesToCode** [IIC2613-Inteligencia-Artificial-2022-2/Syllabus#76]
Hola !

Tengo unos problemas con la parte de los SVM, con lo de entrenar los kernels.

Con un tamano de testeo normal (o sea 20, 30%) se demorra muuuucho (no termina en 15minutos y no tengo mas paciencia que eso ;) )
Con un tamano poco convencional, unos 95% se demora 7minutos pero da un accuracy de 0.70 para el kernel linear por ejemplo.

Tenemos muchos datos asi que puedo entender que se demora mucho... pero es normal ? O deberia haber hecho algo que hace que eso no pasa ?

Aqui es lo que tengo de momento ( viene directamente de lo que esta en las ayudantias) :

from sklearn import datasets
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size = 0.95)

from sklearn import svm
from sklearn.metrics import accuracy_score

linear_classifier = svm.SVC(kernel = "linear")

linear_classifier.fit(X_train, y_train)

accuracy = accuracy_score(y_test, linear_classifier.predict(X_test))

print(accuracy)

También la matriz de confusion se parece a eso 
![image](https://user-images.githubusercontent.com/98588172/200454432-8a01fe45-4d65-490c-a3c4-76e7ea60ec19.png)
que es simplemente no lo que debemos obtener... 

Estoy un poco perdida, creo que no entiendo muy bien como hacer un svm con tantas categorias... si tienen algunos consejos seria lo mejor !

**Comment by cewbeall** [comment]
Estoy teniendo el mismo problema, entonces para añadir a esta pregunta, veo que en el enunciado dice que puede ser que la ejecución sea  más lenta de lo que esperas, ¿pero cuánto debe demorar a lo máximo? Con lo que he probado, aunque el test_size no debe ser tan grande, un test_size menor de 0.95 (y entonces un training size mayor de 0.05) nunca ha terminado su entrenamiento cuando corre.

**Comment by tcouso** [comment]
Hola!

Efectivamente es normal que el modelo se demore mucho tiempo en entrenar, en especial si consideras la gran cantidad de instancias y la alta dimensionalidad de cada una. Parte de los objetivos de la tarea es que puedas lidiar con eso justificando las decisiones que tomes. Una opción es simplemente no considerar la totalidad del dataset, sino que una fracción que te permita tener los resultados requeridos en un tiempo razonable (ojo que esto no se refiere a cambiar las proporciones de entrenamiento/testeo, como hiciste, sino que a limitar el tamaño del dataset en sí).

Sobre el output de tu matriz de confusión, me da la impresión de que es un problema en la renderización del gráfico, por lo que probaría agrandar las dimensiones para que se vea mejor.

Saludos!

**Comment by sebaterrazas** [comment]
Para arreglar el tema de la matriz de confusión se puede agrandar como lo explican en la respuesta a esta pregunta en [https://stackoverflow.com/questions/66483409/adjust-size-of-confusionmatrixdisplay-scikitlearn](StackOverflow), pero según yo se ve mejor si solamente se le saca los números dentro de cada casilla llamando a `plot` de esta manera: `.plot(include_values=False)`
